### DISCUSSION TITLE
why std::vector that is never resized?

### DISCUSSION THREAD

**Discussion by xBZZZZ** [rui314/mold#1378] [General]
if I replace
https://github.com/rui314/mold/blob/2a78b1b3337eacf4f9299f314c1d1b3e2807bd8f/src/icf.cc#L518
with
```cpp
std::array<std::vector<Digest>,3> digests;
```
mold compiles

why `std::vector` used in original code?

**Comment by rui314** [comment]
Or you could write it as `std::vector<Digest> digests[3]`. But that tiny difference doesn't matter, as all of them work fine.
